recipe/patches/0001-Add-Windows-build.patch

```diff
@@ -0,0 +1,296 @@
+From 26b3f4ef3e8c8672e880a09bf474bac5b8cebf80 Mon Sep 17 00:00:00 2001
+From: Michael McAuliffe <michael.e.mcauliffe@gmail.com>
+Date: Sat, 23 Jul 2022 23:48:45 -0700
+Subject: [PATCH] Add Windows build
+
+---
+ extensions/cdrewrite.h | 54 +++++++++++++++++++++---------------------
+ extensions/cpynini.pxd |  4 ++--
+ extensions/getters.cc  | 10 ++++----
+ setup.py               | 32 ++++++++++++++++++++-----
+ 4 files changed, 60 insertions(+), 40 deletions(-)
+
+diff --git extensions/cdrewrite.h extensions/cdrewrite.h
+index 0322f1e..ee094d3 100644
+--- extensions/cdrewrite.h
++++ extensions/cdrewrite.h
+@@ -49,9 +49,9 @@
+ 
+ namespace fst {
+ 
+-enum CDRewriteDirection { LEFT_TO_RIGHT, RIGHT_TO_LEFT, SIMULTANEOUS };
++enum class CDRewriteDirection { LEFT_TO_RIGHT, RIGHT_TO_LEFT, SIMULTANEOUS };
+ 
+-enum CDRewriteMode { OBLIGATORY, OPTIONAL };
++enum class CDRewriteMode { MODE_OBLIGATORY, MODE_OPTIONAL };
```

Comment:
I'm happy to do a `post` release where these become enum classes. What's the point of `MODE_` prefix, is there something else in the namespace conflicting?

Reply:
Yeah, plain "OPTIONAL" throws a syntax error:

```
C:\Users\michael\miniconda3\conda-bld\pynini_1658677198207\work\extensions\cdrewrite.h(419): error C2760: syntax error: unexpected token ':', expected 'expression'
  C:\Users\michael\miniconda3\conda-bld\pynini_1658677198207\work\extensions\cdrewrite.h(581): error C2760: syntax error: unexpected token ':', expected 'expression'
```

Reply:
Wow, trashy.

Reply:
Yeah, the only reference I can find is to "std::optional", but I don't think that would trigger this? Building on Windows is always an adventure...